yarn-project/prover-client/src/prover-agent/memory-proving-queue.test.ts

```diff
@@ -42,6 +47,32 @@ describe('MemoryProvingQueue', () => {
     expect(job2?.request.type).toEqual(ProvingRequestType.BASE_ROLLUP);
   });
 
+  it('returns jobs ordered by priority', async () => {
+    // We push base rollup proof requests for a first block
+    void queue.getBaseRollupProof(makeBaseRollupInputs(), undefined, 1);
+    void queue.getBaseRollupProof(makeBaseRollupInputs(), undefined, 1);
+
+    // The agent consumes one of them
+    expect((await queue.getProvingJob())!.request.type).toEqual(ProvingRequestType.BASE_ROLLUP);
+
+    // A new block comes along with its base rollups, and the orchestrator then pushes a root request for the first one
+    void queue.getBaseRollupProof(makeBaseRollupInputs(), undefined, 2);
+    void queue.getBaseRollupProof(makeBaseRollupInputs(), undefined, 2);
+    void queue.getBaseRollupProof(makeBaseRollupInputs(), undefined, 2);
+    void queue.getBaseRollupProof(makeBaseRollupInputs(), undefined, 2);
+    void queue.getRootRollupProof(makeRootRollupInputs(), undefined, 1);
+
+    // The next jobs for the agent should be the ones from block 1, skipping the ones for block 2
+    expect((await queue.getProvingJob())!.request.type).toEqual(ProvingRequestType.BASE_ROLLUP);
+    expect((await queue.getProvingJob())!.request.type).toEqual(ProvingRequestType.ROOT_ROLLUP);
```

Comment:
I'd make the `epochNumber` available on the proving job. It will be helpful to log it alongside proving job type in the agent